File: services/notification-service/src/common/configs/rabbitmqconstant.config.ts
class RabbitMQConstantConfig {
  // ---------- Queue constant ----------
  private notificationServiceQueue: string = 'notification-service-queue';

  // ---------- Exchange constant ----------
  private forceLoggedOutExchange: string = 'force-logged-out';

  // ---------- Routing key constant ----------
  private forceLoggedOutRoutingKey: string = 'force-logged-out';

  public getNotificationServiceQueue(): string {
    return this.notificationServiceQueue;
  }

  public getForceLoggedOutExchange(): string {
    return this.forceLoggedOutExchange;
  }

  public getForceLoggedOutRoutingKey(): string {
    return this.forceLoggedOutRoutingKey;
  }
}

export const rabbitMQConstantConfig: RabbitMQConstantConfig =
  new RabbitMQConstantConfig();

File: services/identity-service/src/main/java/com/zwind/identityservice/constant/RabbitMQConstantConfig.java
package com.zwind.identityservice.constant;

import org.springframework.stereotype.Component;

@Component
public class RabbitMQConstantConfig {
    // ---------- Queue constant ----------
    public static final String PROFILE_EVENTS_QUEUE = "q.profile.events";
    public static final String USER_DEAD_LETTER_QUEUE = "q.user.dlq";

    // ---------- Exchange constant ----------
    public static final String IDENTITY_SERVICE_EXCHANGE = "x.service.identity";
    public static final String PROFILE_EVENTS_EXCHANGE = "x.profile.events";
    public static final String USER_DEAD_LETTER_EXCHANGE = "x.user.dlx";

    // ---------- Routing key constant ----------
    public static final String USER_CREATED_RK = "user.created";
    public static final String PROFILE_CREATED_STATUS_RK = "profile.created.*";
    public static final String USER_DEAD_RK = "user.dead";

    private RabbitMQConstantConfig() {}
}
